crates/sui-move/tests/new_tests.rs

```diff
@@ -0,0 +1,385 @@
+    # override use `override = true`
+    # Override = { local = "../conflicting/version", override = true }
+
+    [addresses]
+    example = "0x0"
+
+    # Named addresses will be accessible in Move as `@name`. They're also exported:
+    # for example, `std = "0x1"` is exported by the Standard Library.
+    # alice = "0xA11CE"
+
+    [dev-dependencies]
+    # The dev-dependencies section allows overriding dependencies for `--test` and
+    # `--dev` modes. You can introduce test-only dependencies here.
+    # Local = { local = "../path/to/dev-build" }
+
+    [dev-addresses]
+    # The dev-addresses section allows overwriting named addresses for the `--test`
+    # and `--dev` modes.
+    # alice = "0xB0B"
+    "###);
+}
+
+#[test]
+#[cfg(feature = "unit_test")]
+/// `sui-move build` and `sui-move test` succeed on a newly generated project
+fn test_new_build() {
+    let (proj, mut cmd) = sui_move();
+
+    // sui move new example
+    assert_cmd_snapshot!(cmd.arg("new").arg("example"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    "###);
+
+    // Check that generated examples can build
+    let (_, mut build) = sui_move();
+    assert_cmd_snapshot!(build.current_dir(proj.path().join("example")).arg("build"), @"TODO: this test should run!");
+
+    // Check that generated example tests can run
+    let (_, mut test) = sui_move();
+    assert_cmd_snapshot!(test.current_dir(proj.path().join("example")).arg("test"), @"TODO: this test should run!");
+}
+
+#[test]
+/// `sui-move new example` when `example/.gitignore` exists: it should be modified rather than
+/// replaced
+fn test_new_gitignore_exists() {
+    let (mut proj, mut cmd) = sui_move();
+
+    // create .gitignore file
+    new_file(&mut proj, "example/.gitignore", "existing_ignore\n").unwrap();
+
+    // sui move new example
+    assert_cmd_snapshot!(cmd.arg("new").arg("example"), @r###"
+    success: true
+    exit_code: 0
+    ----- stdout -----
+
+    ----- stderr -----
+    "###);
+
+    // check list of files
+    assert_debug_snapshot!(recursive_paths(proj.path().join("example")), @r###"
+    [
+        "",
+        ".gitignore",
+        "Move.toml",
+        "sources",
+        "sources/example.move",
+        "tests",
+        "tests/example_tests.move",
+    ]
+    "###);
+
+    // check .gitignore contents
+    assert_snapshot!(slurp_file(&mut proj, "example/.gitignore").expect("sui move new updates .gitignore"), @r###"
+    existing_ignore
+    build/*
+    "###);
+}
+
+#[test]
+/// `sui-move new example` when `example/.gitignore` already contains `build/*`: it should be
+/// unchanged
+fn test_new_gitignore_has_build() {
+    let (mut proj, mut cmd) = sui_move();
+
+    // create .gitignore file containing `build/*`
+    new_file(
+        &mut proj,
+        "example/.gitignore",
+        r###"
+first_ignore
+build/*
+another_ignore
+        "###,
```

Comment:
nit (optional): This may flow more nicely?

```suggestion
        "first_ignore\n\
         build/*\n\
         another_ignore",
```